docker/Dockerfile.hardhat

```diff
@@ -11,18 +11,22 @@ RUN npm install --ignore-scripts
 
 COPY ./hardhat .
 
-# Change ownership of the app directory to the non-root user
-RUN chown -R hardhat-user:hardhat-group /app
+# Create directories and set permissions BEFORE switching user
+RUN mkdir -p /app/deployments/localhost && \
+    mkdir -p /app/deployments/hardhat && \
+    mkdir -p /app/artifacts/build-info && \
+    mkdir -p /app/cache && \
+    chown -R hardhat-user:hardhat-group /app && \
+    chmod -R 777 /app/deployments && \
+    chmod -R 777 /app/artifacts && \
+    chmod -R 777 /app/cache
 
 ENV PATH="/app/node_modules/.bin:${PATH}"
 
-RUN echo -e '#!/bin/sh\necho "Compiling contracts..."\nnpx hardhat compile --force\necho "Starting Hardhat node..."\nexec ./node_modules/.bin/hardhat node --network hardhat' > /app/start.sh && \
-    chmod +x /app/start.sh && \
-    ls -la /app/start.sh
+# Copy the start script from hardhat/scripts instead of creating it
+COPY ./hardhat/scripts/start.sh /app/start.sh
+RUN chmod +x /app/start.sh
```

Comment:
@mpaya5 in the past we have had many problems executing shell scripts when the studio runs on Windows. Is there a way to test this on your side?

Reply:
You were right. I pushed a new commit solving that: https://github.com/yeagerai/genlayer-studio/pull/670/commits/a9ab9afe02b270827beab19ba36fb77da4c9ad59

Reply:
@mpaya5 what was the issue? I'm curious

Reply:
The error was  that it couldn’t find the script.sh file on Windows. I’m not sure why this happened. This occurred after running `docker compose up.`